lib/phoenix/pubsub/pg2_server.ex

```diff
@@ -24,11 +24,15 @@ defmodule Phoenix.PubSub.PG2Server do
     end
   end
 
-  def init(name) do
-    pg2_namespace = pg2_namespace(name)
-    :ok = :pg2.create(pg2_namespace)
-    :ok = :pg2.join(pg2_namespace, self)
-    {:ok, name}
+  def init({server_name, node_name}) do
+    public_pg2_namespace = pg2_namespace(server_name, :global)
+    private_pg2_namespace = pg2_namespace(server_name, node_name)
+    :ok = :pg2.create(public_pg2_namespace)
+    :ok = :pg2.create(private_pg2_namespace)
+    :ok = :pg2.join(public_pg2_namespace, self)
+    :ok = :pg2.join(private_pg2_namespace, self)
```

Comment:
Because pg2 relies on a fully connected Erlang network, we theoretically don't need a private pg2. We can just send a message using `send {registered_name, node}, message`. Thoughts?